### PR TITLE
Fix d9a04ba446: Ensure all MD5Hash are initialized.

### PR DESCRIPTION
**Pull request by PeterN**
## Motivation / Problem

In some places an MD5Hash is xored with another hash, without ensure the original array is clear. This results in false MD5 mismatches.

## Description

Ensure all MD5Hashes are value initialized. Not all instances need to be initialized as often they are copied or written to, but doing all ensures no surprises.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

**Review by glx22 — DISMISSED** [review]

**Comment by TrueBrain** [comment]
I mentioned it on Discord, but this feels like this will go wrong a lot more often in the future. It isn't all that clear you need to initialize MD5Hash. Makes me wonder if we shouldn't just make it a class and put an explicit ctor on it. But that might just be me being annoyed by PODs and C++ :P

**Comment by PeterN** [comment]
Okay, as @TrueBrain mentioned, I have now replaced the original commit by this which gives MD5Hash a default constructor that value initializes itself, so anyone using MD5Hash doesn't need to care about the underlying container.

**Comment by glx22** [comment]
I think it would make sense to also put the xor operator inside the struct.

**Comment by TrueBrain** [comment]
I would say that comment, well, both of them, aged like milk :D :D :D :D :)

(sorry, couldn't resist poking fun :P)

File: src/3rdparty/md5/md5.h [hunk 1]
```diff
@@ -57,20 +57,20 @@
...
-	for (size_t i = 0; i < lhs.size(); i++) lhs[i] ^= rhs[i];
-	return lhs;
-}
-
+struct MD5Hash : std::array<byte, MD5_HASH_BYTES> {
+	MD5Hash() : std::array<byte, MD5_HASH_BYTES>{} {}
+
+	/**
+	 * Exclusively-or the given hash into this hash.
+	 * @param other The other has.
```
Comment (TrueBrain) [inline src/3rdparty/md5/md5.h]: has? HASH! :)

Comment (PeterN) [reply src/3rdparty/md5/md5.h]: I can't even just copy & paste Rubidium's code without getting critiqued! ;-)

**Review by TrueBrain — APPROVED** [review]

**Pull request merged**